client/lib/wp/localization/index.js

```diff
@@ -43,9 +43,17 @@ export function addLocaleQueryParam( params ) {
 		return params;
 	}
 
+	let localeQueryParam = {};
 	const query = qs.parse( params.query );
+
+	if ( 'wpcom/v2' === params.apiNamespace ) {
```

Comment:
> In which case it might be fair enough to simply test truthiness of `params.apiNamespace` to decide to prefix the underscore.

Yes, this is correct.